src/HandlerRequestContext.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Ingenerator\ApiEmulator;
+
+class HandlerRequestContext
+{
+
+    public function __construct(
+        public readonly HandlerDataRepository $data_repository
```

Comment:
Thinking the layer of indirection here (handler -> HandlerRequestContext -> data_repository) gives us the option to add more properties in future, so we can provide additional stuff to handlers without breaking BC and while keeping the function signature for the handler itself simple.